ksmt-cvc5/src/main/kotlin/org.ksmt.solver.cvc5/KCvc5Model.kt

```diff
@@ -0,0 +1,135 @@
+package org.ksmt.solver.cvc5
+
+import io.github.cvc5.Term
+import org.ksmt.KContext
+import org.ksmt.decl.KConstDecl
+import org.ksmt.decl.KDecl
+import org.ksmt.decl.KFuncDecl
+import org.ksmt.expr.KExpr
+import org.ksmt.expr.rewrite.KExprUninterpretedDeclCollector
+import org.ksmt.solver.KModel
+import org.ksmt.solver.model.KModelImpl
+import org.ksmt.sort.KBoolSort
+import org.ksmt.sort.KSort
+import org.ksmt.sort.KUninterpretedSort
+import org.ksmt.utils.mkFreshConst
+
+open class KCvc5Model(
+    private val ctx: KContext,
+    private val cvc5Ctx: KCvc5Context,
+    private val assertions: Collection<KExpr<KBoolSort>>,
+    private val converter: KCvc5ExprConverter,
+    private val internalizer: KCvc5ExprInternalizer
+) : KModel {
+
+    override val declarations: Set<KDecl<*>> by lazy {
+        buildSet {
+            assertions.forEach { assertion ->
+                this += KExprUninterpretedDeclCollector.collectUninterpretedDeclarations(assertion)
+            }
+        }
+    }
+
+    override val uninterpretedSorts: Set<KUninterpretedSort> by lazy {
+        buildSet {
+            assertions.forEach { assertion ->
+                this += KExprUninterpretedSortCollector.collectUninterpretedSorts(assertion)
+            }
+        }
+    }
+
+    private val interpretations = hashMapOf<KDecl<*>, KModel.KFuncInterp<*>?>()
+    private val uninterpretedSortsUniverses = hashMapOf<KUninterpretedSort, Set<KExpr<KUninterpretedSort>>>()
+
+    /*
+    Always returns evaluated value regardless of isComplete (forever isComplete = true).
```

Comment:
Please, add some clarification on the eval behaviour in a case when expr contains declarations that were previously unknown to the solver.